src/main/java/core/basesyntax/strategy/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Operation, OperationHandler> strategiesMap;
+
+    public OperationStrategyImpl(Map<Operation, OperationHandler> strategiesMap) {
+        this.strategiesMap = strategiesMap;
+    }
+
+    @Override
+    public OperationHandler getHandler(Operation operation) {
+        String exceptionMessage =
+                "Cannot find OperationHandler for operation = [" + operation + "]";
+        return Optional.ofNullable(strategiesMap.get(operation))
+                .orElseThrow(() -> new NoSuchElementException(exceptionMessage));
```

Comment:
According to the checklist, when handling exceptions, you should rethrow a RuntimeException with an informative message and exception object. In this case, it's better to throw a RuntimeException instead of NoSuchElementException to keep the exception handling consistent across the project.